liblangutil/Exceptions.h

```diff
@@ -139,21 +147,71 @@ class Error: virtual public util::Exception
 	static Error const* containsErrorOfType(ErrorList const& _list, Error::Type _type)
 	{
 		for (auto e: _list)
-		{
 			if (e->type() == _type)
 				return e.get();
-		}
 		return nullptr;
 	}
-	static bool containsOnlyWarnings(ErrorList const& _list)
+
+	static Category errorCategory(Type _type)
+	{
+		if (_type == Type::Info)
+			return Category::Info;
+		if (_type == Type::Warning)
+			return Category::Warning;
+		return Category::Error;
+	}
+
+	static bool isError(Category _cat)
+	{
+		return _cat == Category::Error;
+	}
+
+	static bool isError(Type _type)
+	{
+		return isError(errorCategory(_type));
+	}
+
+	static bool containsErrors(ErrorList const& _list)
 	{
 		for (auto e: _list)
-		{
-			if (e->type() != Type::Warning)
-				return false;
-		}
-		return true;
+			if (isError(e->type()))
+				return true;
+		return false;
 	}
+
+	static std::string formatErrorCategory(Category _cat)
+	{
+		if (_cat == Category::Info)
+			return "Info";
+		if (_cat == Category::Warning)
+			return "Warning";
+		solAssert(isError(_cat), "");
+		return "Error";
+	}
+
+	static std::string formatErrorCategoryLowercase(Category _cat)
+	{
+		if (_cat == Category::Info)
```

Comment:
`Category` is an enum, why not just have a switch statement here? We do have the compiler warning enabled for missing switch cases.